packages/liveblocks-core/src/room.ts

```diff
@@ -1230,6 +1256,11 @@ export type RoomConfig = {
 
   baseUrl: string;
   enableDebugLogging?: boolean;
+
+  // We would not have to pass this complicated factory/callback functions to
+  // the createRoom() function if we would simply pass the Client instance to
+  // the Room instance, so it can directly call this back on the Client.
+  createSyncSource: () => SyncSource;
```

Comment:
I see that we still pass this factory to the createRoom function. I remember us discussing potentially doing this directly within the client. Do you still plan to do that or was it not easily possible to do it from client after further look?

Reply:
I tried it, but it turned out to be harder to do. I needed to add quite a bit of extra complexity around tearing down the room and Storage/Yjs listeners in the leave room exit function, so I decided to keep this implementation instead. I still think that conceptually it should be possible to do this, but the current implementation made it harder than I wanted it to be.